### PR TITLE
Converge semgrep metrics

### PR DESCRIPTION
**Pull request by dinosaure**
PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [ ] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

This PR wants to converge `semgrep_metrics.atd` between the Python code and the OCaml code. Currently, this PR requires returntocorp/semgrep-interfaces#71. I tried to find the best way to fit into the OCaml and the Python code without big changes. However (/cc @mjambon), it seems we can not express with `atdpy` that some values have defaults on the Python side but must be present on the JSON side.

Indeed, one test fails where some unset values are not output by the Python JSON encoder. It will be nice to:
1) be able to express that via `atdpy`
2) or update the test (and what we expects about the JSON output)

I marked this PR as a draft due to the pending PR about `semgrep-interfaces`.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -100,3 +100,5 @@
 [submodule "languages/cairo/tree-sitter/semgrep-cairo"]
 	path = languages/cairo/tree-sitter/semgrep-cairo
 	url = https://github.com/returntocorp/semgrep-cairo
+[submodule "cli/src/semgrep/semgrep_interfaces"]
```
Comment (aryx) [inline .gitmodules]: This will need to be removed

**Review by aryx — COMMENTED** [review]
Very nice! I like it.

**Comment by dinosaure** [comment]
Currently, tests still fails because one test `test_metrics_payload[pro_flag0]` expects to **not** have the `maxMemoryBytes` field and `test_metrics_payload[pro_flag1]` expects to have this field (even if it's not set - with the `null` value). So we must do a choice on that and transform (as I did on this PR) the `maxMemoryBytes` field as an optional field (and upgrade the `test_metrics_payload[pro_flag1]` accordingly) or define it as a required field (and upgrade the `test_metrics_payload[pro_flag0]` accordingly).

**Comment by aryx** [comment]
I wonder how the original python code was working to allow both situations ... weird.

File: src/osemgrep/core/Metrics_.ml [hunk 2]
```diff
@@ -71,6 +71,14 @@ let default_payload =
         profilingTimes = [];
         maxMemoryBytes = None;
       };
+    extension =
```
Comment (aryx) [inline src/osemgrep/core/Metrics_.ml]: BTW could you try to rename Metrics_.ml to just Metrics.ml
or maybe it goes in conflict with some library we use that already have a Metrics.ml, but in any case
it's worth adding  a comment in the .ml to say why it's not called more simply just Metrics.ml

**Review by aryx — CHANGES_REQUESTED** [review]

File: src/osemgrep/core/Metrics_.ml (hunk 2 above)
Comment (dinosaure) [reply src/osemgrep/core/Metrics_.ml]: I think @hannesm renamed it due to the usage of the `metrics` library - a clash occurs between the library and this module. I can add a note indeed.

**Comment by aryx** [comment]
remove the Draft thing when you think this is ready for review @dinosaure

**Review by aryx — APPROVED** [review]
This is great

**Comment by aryx** [comment]
I think you need to remove the old Semgrep_metrics.atd now that we use the one in semgrep-interfaces/ no?
I don't see the file deleted in the PR.

**Review by aryx — APPROVED** [review]

**Pull request merged**